src/httpserver.cpp

```diff
@@ -639,6 +641,39 @@ HTTPRequest::RequestMethod HTTPRequest::GetRequestMethod() const
     }
 }
 
+std::optional<std::string> HTTPRequest::GetQueryParameter(const std::string& key) const
+{
+    auto uri = evhttp_request_get_uri(req);
+    auto result = GetQueryParameterFromUri(uri, key);
+
+    return result;
+}
+
+std::optional<std::string> GetQueryParameterFromUri(const char* uri, const std::string& key)
+{
+    auto uri_parsed = evhttp_uri_parse(uri);
+    auto query = evhttp_uri_get_query(uri_parsed);
+    std::optional<std::string> result {};
```

Comment:
Thanks, I wasn't aware of this convention. I've updated this and 3 other zero-inits of `std::string`. I'm still zero-init'ing `size_t blockhash_index {};` as to my understanding `long` doesn't have a default ctor and this could be unsafe otherwise.